pkgs/tools/networking/p2p/tahoe-lafs/default.nix

```diff
@@ -1,65 +1,60 @@
-
-# FAILURES: The "running build_ext" phase fails to compile Twisted
-# plugins, because it tries to write them into Twisted's (immutable)
-# store path. The problem appears to be non-fatal, but there's probably
-# some loss of functionality because of it.
+{ fetchFromGitHub, lib, unzip, nettools, pythonPackages, texinfo, sqlite }:
 
 pythonPackages.buildPythonApplication rec {
-  version = "1.13.0";
+  version = "unstable-2019-04-08";
   name = "tahoe-lafs-${version}";
   namePrefix = "";
 
-  src = fetchurl {
-    url = "https://tahoe-lafs.org/downloads/tahoe-lafs-${version}.tar.bz2";
-    sha256 = "11pfz9yyy6qkkyi0kskxlbn2drfppx6yawqyv4kpkrkj4q7x5m42";
+  src = fetchFromGitHub {
+    owner = "tahoe-lafs";
+    repo = "tahoe-lafs";
+    rev = "325c522d7c73065e3f7e10518a9785b822d1405a";
+    sha256 = "0000000000000000000000000000000000000000000000000000";
   };
 
   outputs = [ "out" "doc" "info" ];
 
-  postPatch = ''
-    sed -i "src/allmydata/util/iputil.py" \
-        -es"|_linux_path = '/sbin/ifconfig'|_linux_path = '${nettools}/bin/ifconfig'|g"
-
-    # Chroots don't have /etc/hosts and /etc/resolv.conf, so work around
-    # that.
-    for i in $(find src/allmydata/test -type f)
-    do
-      sed -i "$i" -e"s/localhost/127.0.0.1/g"
-    done
-
-    sed -i 's/"zope.interface.*"/"zope.interface"/' src/allmydata/_auto_deps.py
-    sed -i 's/"pycrypto.*"/"pycrypto"/' src/allmydata/_auto_deps.py
-  '';
-
-  # Remove broken and expensive tests.
-  preConfigure = ''
-    (
-      cd src/allmydata/test
-
-      # Buggy?
-      rm cli/test_create.py test_backupdb.py
-
-      # These require Tor and I2P.
-      rm test_connections.py test_iputil.py test_hung_server.py test_i2p_provider.py test_tor_provider.py
-
-      # Expensive
-      rm test_system.py
-    )
-  '';
-
-  nativeBuildInputs = with pythonPackages; [ sphinx texinfo ];
-
-  buildInputs = with pythonPackages; [ unzip numpy mock ];
+  nativeBuildInputs = with pythonPackages; [
+    sphinx texinfo
+  ];
 
   # The `backup' command requires `sqlite3'.
   propagatedBuildInputs = with pythonPackages; [
-    twisted foolscap nevow simplejson zfec pycryptopp darcsver
-    setuptoolsTrial setuptoolsDarcs pycrypto pyasn1 zope_interface
-    service-identity pyyaml magic-wormhole treq
+    sqlite
+    characteristic
+    eliot
+    foolscap
+    magic-wormhole
+    nevow
+    pyasn1
+    pyasn1-modules
+    pycryptopp
+    pyopenssl
+    pyyaml
+    service-identity
+    twisted
+    twisted.extras.conch
+    twisted.extras.tls
+    zfec
+  ];
+
+  checkInputs = with pythonPackages; [
+    certifi
+    subunitreporter
+    hypothesis
+    treq
+    testtools
+    fixtures
   ];
 
-  checkInputs = with pythonPackages; [ hypothesis twisted ];
+  # Tests are usually run via tox, but we don't need this overhead.
+  # Still a lot of tests are failing, need some help to debug this.
+  # checkPhase = ''
+  #   trial --rterrors allmydata
+  # '';
```

Comment:
Please keep the `checkPhase` and set `doCheck = false`.
We might be able to help if you post the output of a test run.